src/bam/record.rs

```diff
@@ -769,8 +769,8 @@ pub enum Cigar {
 }
 
 impl Cigar {
-    fn encode(&self) -> u32 {
-        match *self {
+    fn encode(self) -> u32 {
```

Comment:
Interesting! Are you sure this is idiomatic? Semantically, I would think a borrow makes more sense here.